lib/mayaUsd/nodes/layerManager.cpp

```diff
@@ -159,27 +159,19 @@ void convertAnonymousLayersRecursive(
 {
     auto currentTarget = stage->GetEditTarget().GetLayer();
 
-    MayaUsd::utils::LayerParent parentPtr;
-    if (stage->GetRootLayer() == layer) {
-        parentPtr._layerParent = nullptr;
-        parentPtr._proxyPath = basename;
-    } else if (stage->GetSessionLayer() == layer) {
-        parentPtr._layerParent = nullptr;
-        parentPtr._proxyPath = basename;
-    } else {
-        parentPtr._layerParent = layer;
-        parentPtr._proxyPath = basename;
-    }
-
```

Comment:
The code above was creating the parent layer description of the given layer, but this was used to describe the parent of the sub-layer! So when processing the sub-layer, it was using its grand-parent instead. This led to the confusing behaviour that an anonymous sub-layer of the root would become the root.